packages/utils/src/ast.test.ts

```diff
@@ -0,0 +1,261 @@
+  });
+
+  it(`doesn't attach comments to the AST if configured`, () => {
+    const code = `
+      // This is a comment.
+      const foo = 'bar';
+    `;
+
+    const ast = getAST(code, false);
+    expect(getCode(ast)).not.toContain('// This is a comment.');
+  });
+
+  it('forwards parser errors', () => {
+    // Invalid code
+    const code = `const foo bar;`;
+
+    expect(() => getAST(code)).toThrow(
+      `Failed to parse the provided code to an AST:`,
+    );
+  });
+});
+
+describe('postProcessAST', () => {
+  it('wraps eval', () => {
+    const code = `
+      eval(bar);
+      foo.eval(bar, baz);
+      foo.bar.eval(baz, qux);
+      eval('<!-- foo -->');
+      foo.eval('<!-- bar -->');
+    `;
+
+    const ast = getAST(code);
+    const processedCode = postProcessAST(ast);
+
+    expect(getCode(processedCode)).toMatchInlineSnapshot(`
+      "(1, eval)(bar);
+      (1, foo.eval)(bar, baz);
+      (1, foo.bar.eval)(baz, qux);
+      (1, eval)(\\"<!\\" + \\"--\\" + \\" foo \\" + \\"--\\" + \\">\\");
+      (1, foo.eval)(\\"<!\\" + \\"--\\" + \\" bar \\" + \\"--\\" + \\">\\");"
+    `);
+  });
+
+  it('removes the Buffer argument', () => {
+    const code = `
+      (function (Buffer, foo) {
+        const bar = 'baz';
+      });
+
+      function foo(Buffer) {
+        const bar = 'baz';
+      }
+    `;
+
+    const ast = getAST(code);
+    const processedCode = postProcessAST(ast);
+
+    expect(getCode(processedCode)).toMatchInlineSnapshot(`
+      "(function (foo) {
+        const bar = \\"baz\\";
+      });
+
+      function foo(Buffer) {
+        const bar = \\"baz\\";
+      }"
+    `);
+  });
+
+  it('inserts the regenerator runtime global when used', () => {
+    const code = `
+      function foo() {
+        regeneratorRuntime.foo();
+      }
+
+      function bar() {
+        regeneratorRuntime.bar();
+      }
+    `;
+
+    const ast = getAST(code);
+    const processedCode = postProcessAST(ast);
+
+    expect(getCode(processedCode)).toMatchInlineSnapshot(`
+      "var regeneratorRuntime;
+
+      function foo() {
+        regeneratorRuntime.foo();
+      }
+
+      function bar() {
+        regeneratorRuntime.bar();
+      }"
+    `);
+  });
+
+  it('does not insert the regenerator runtime global when not used', () => {
+    const code = `
+      const foo = 'regeneratorRuntime';
+    `;
+
+    const ast = getAST(code);
+    const processedCode = postProcessAST(ast);
+
+    expect(getCode(processedCode)).toMatchInlineSnapshot(
+      `"const foo = \\"regeneratorRuntime\\";"`,
+    );
+  });
+
+  it('breaks up HTML comment terminators in operators', () => {
+    const code = `
+      let a = 5;
+      let b = 7;
+      const c = a --> b;
+      const d = a <!-- b;
+    `;
+
+    const ast = getAST(code);
+    const processedCode = postProcessAST(ast);
+
+    expect(getCode(processedCode)).toMatchInlineSnapshot(`
+      "let a = 5;
+      let b = 7;
+      const c = a-- > b;
+      const d = a < ! --b;"
```

Comment:
We should make sure this is what we want before merging the final PR. Perhaps we should reach out to Agoric to hear their thoughts on it.

Reply:
Now that we have the control to handle it properly 😉